app/src/main/java/com/adammcneilly/toa/core/ui/components/navigation/TOANavigationDrawerContent.kt

```diff
@@ -0,0 +1,108 @@
+package com.adammcneilly.toa.core.ui.components.navigation
+
+import android.content.res.Configuration
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.sizeIn
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.Icon
+import androidx.compose.material3.NavigationDrawerItem
+import androidx.compose.material3.PermanentDrawerSheet
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.navigation.NavHostController
+import com.adammcneilly.toa.ExcludeFromJacocoGeneratedReport
+import com.adammcneilly.toa.core.ui.theme.TOATheme
+import com.ramcosta.composedestinations.utils.currentDestinationAsState
+
+@Composable
+fun TOANavigationDrawerContent(
+    navHostController: NavHostController,
+    tabs: List<NavigationTab>,
+    modifier: Modifier = Modifier,
+) {
+    val currentRoute = navHostController.currentDestinationAsState().value?.route
+
+    val shouldShowNavigationDrawer = tabs.any { tab ->
+        tab.screenRoute == currentRoute
+    }
+
+    if (shouldShowNavigationDrawer) {
```

Comment:
This logic is duplicated in all three navigation flows, maybe we can pull it out into its own function?